sdk/keyvault/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,141 @@
+* [Troubleshooting Azure Key Vault Secrets SDK Issues](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Secrets/TROUBLESHOOTING.md)
+
+See [Azure SDK diagnostics](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md)
+for help diagnosing various problems across all our Azure SDKs for .NET.
+
+## Table of Contents
+
+* [Troubleshooting Authentication Issues](#troubleshooting-authentication-issues)
+  * [HTTP 401 Errors](#http-401-errors)
+    * [Frequent HTTP 401 Errors in Logs](#frequent-http-401-errors-in-logs)
+    * [AKV10032: Invalid issuer](#akv10032-invalid-issuer)
+    * [Other Authentication Issues](#other-authentication-issues)
+  * [HTTP 403 Errors](#http-403-errors)
+    * [Operation Not Permitted](#operation-not-permitted)
+    * [Access Denied to First Party Service](#access-denied-to-first-party-service)
+* [Other Service Errors](#other-service-errors)
+  * [HTTP 429: Too Many Request](#http-429-too-many-requests)
+
+## Troubleshooting Authentication Issues
+
+### HTTP 401 Errors
+
+HTTP 401 errors may indicate problems authenticating.
+
+#### Frequent HTTP 401 Errors in Logs
+
+Most often, this is expected. Azure Key Vault issues a challenge for initial requests that force authentication. You may
+see these errors most often during application startup, but may also see these periodically during the application's
+lifetime when authentication tokens are near expiration.
+
+If you are not seeing subsequent exceptions from the Key Vault SDKs, authentication challenges are likely the cause.
+
+#### AKV10032: Invalid issuer
+
+You may see an error similar to:
+
+```text
+Azure.RequestFailedException : Service request failed.
+Status: 401 (Unauthorized)
+
+Content:
+{"error":{"code":"Unauthorized","message":"AKV10032: Invalid issuer. Expected one of https://sts.windows.net/{tenant 1}/, found https://sts.windows.net/{tenant 2}/."}}
+```
+
+This is most often caused by being logged into a different tenant than the Key Vault authenticates.
+See our [DefaultAzureCredential] documentation to see the order credentials are read. You may be logged into a different
+tenant for one credential that gets read before another credential. For example, you might be logged into Visual Studio
+under the wrong tenant even though you're logged into the Azure CLI under the right tenant.
+
+#### Other Authentication Issues
+
+If you are using the `Azure.Identity` package - which contains [DefaultAzureCredential] - to authenticate requests to
+Azure Key Vault, please visit our [troubleshooting guide](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/TROUBLESHOOTING.md).
+
+### HTTP 403 Errors
+
+HTTP 403 errors indicate the user is not authorized to perform a specific operation in Key Vault or Managed HSM.
```

Comment:
This should probably mention that the user should confirm they are using the intended account, and contain links to more information about configuring KV/MHSM permissions.